examples/bpi-monolith-test-suite/bpi/components/zkmock/shield.ts

```diff
@@ -1,30 +1,36 @@
 import { Verifier } from "./verifier";
+import { getProof, verifyAndAddNewLeaf } from "./blockchain/shieldUtils";
 
 export class Shield {
-    verifiers: { [id: string]: Verifier; } = {};
-    agreementStateCommitment: Buffer;
-    id: string;
-    owner: string;
+  verifiers: { [id: string]: Verifier } = {};
+  agreementStateCommitment: Buffer;
+  id: string;
+  owner: string;
 
-    constructor(id: string, initState: Buffer, owner: string) {
-        this.id = id;
-        this.agreementStateCommitment = initState;
-        this.owner = owner
-    }
+  constructor(id: string, initState: Buffer, owner: string) {
+    this.id = id;
+    this.agreementStateCommitment = initState;
+    this.owner = owner;
+  }
 
-    addVerifier(verifierId: string, verifier: Verifier, callingAddress: string) {
-        if (callingAddress === this.owner) { // In practice contract would inherit from owned, calling address would be msg.sender
-            this.verifiers[verifierId] = verifier;
-        }
-    }
-    executeWorkstep(verifierId: string, stateObjectCommitment: Buffer, privateInputs: Object): any {
-        let proof = privateInputs;
-        let input = {
-            agreementStateCommitment: this.agreementStateCommitment,
-            stateObjectCommitment: stateObjectCommitment
-        };
-        this.agreementStateCommitment = this.verifiers[verifierId].verifyProof(proof, input);
-        return true
+  addVerifier(verifierId: string, verifier: Verifier, callingAddress: string) {
+    if (callingAddress === this.owner) {
+      // In practice contract would inherit from owned, calling address would be msg.sender
+      this.verifiers[verifierId] = verifier;
     }
+  }
 
-}
\ No newline at end of file
+  async executeWorkstep(stateObjectCommitment: Buffer): Promise<any> {
+    var proof = await getProof();
+    var input = {
+      agreementStateCommitment: this.agreementStateCommitment,
+      stateObjectCommitment: stateObjectCommitment,
+    };
+    var executionStatus = verifyAndAddNewLeaf(
```

Comment:
small thing, but return verifyAndAddNewLeaf.. is better just to be sure this is returned from this method, since it should be awaited, and you don't have await here, but because we will return it it will be awaited when await executeWorkstep is executed